base/collection.go

```diff
@@ -165,11 +165,14 @@ func GetGocbV2BucketFromCluster(ctx context.Context, cluster *gocb.Cluster, spec
 	if mgmtEpsErr != nil && len(mgmtEps) > 0 {
 		nodeCount = len(mgmtEps)
 	}
-	numPools := 1
-	if spec.KvPoolSize > 0 {
-		numPools = spec.KvPoolSize
+
+	numPools, err := spec.GetKvPoolSize()
```

Comment:
It feels like GetKvPoolSize should be a helper function that takes a connString, and doesn't use the spec?  That way we can be certain that the value here matches what's actually getting sent to gocb (on the conn string), and we don't need to have any logic inside GetKvPoolSize around default values, etc.

That would also let us get rid of the test-only getKVConnectionPol [sic], and just use the new connString based function for test verification.

Reply:
I changed this to a generic helper function so I could use it for `kv_pool_size` and `kv_buffer_size`.